### ISSUE TITLE
NullReferenceException in CodeElementCache when the .cs extension is exported from an extension

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#35394]
When I create a VS extension that exports a content type for the .cs extension, VS crashes when a .cs file is opened. The crash is in Microsoft.VisualStudio.CodeSense.Roslyn.CodeElementCache.

Extension code:

```
using System.ComponentModel.Composition;
using Microsoft.VisualStudio.Utilities;

internal static class FileAndContentTypeDefinitions
{
    [Export]
    [Name(&quot;CSharpE&quot;)]
    [BaseDefinition(&quot;CSharp&quot;)]
    internal static ContentTypeDefinition cSharpEContentTypeDefinition;

    [Export]
    [FileExtension(&quot;.cs&quot;)]
    [ContentType(&quot;CSharpE&quot;)]
    internal static FileExtensionToContentTypeDefinition cSharpEFileExtensionDefinition;
}
```

Full extension project is in an attachment.

Exception stack trace:

```
System.NullReferenceException
   at Microsoft.VisualStudio.CodeSense.Roslyn.CodeElementCache+&lt;RebuildAsync&gt;d__9.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at Microsoft.VisualStudio.CodeSense.Editor.Roslyn.CodeElementTagger+&lt;UpdateSnapshotAsync&gt;d__15.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(System.Threading.Tasks.Task)
   at Microsoft.VisualStudio.CodeSense.Editor.Tagger`1+&lt;&gt;c__DisplayClass22_1+&lt;&lt;UpdateSnapshotAsync&gt;b__0&gt;d[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at System.Threading.Tasks.AlmTaskExtensions+&lt;FireAndForget&gt;d__1.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+&lt;&gt;c.&lt;ThrowAsync&gt;b__6_0(System.Object)
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory+SingleExecuteProtector.TryExecute()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory+SingleExecuteProtector+&lt;&gt;c.&lt;.cctor&gt;b__20_0(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
```

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/347756/nullreferenceexception-in-codeelementcache-when-th.html
VSTS ticketId: 695305_
_These are the original issue comments:_

Jinu Joseph [MSFT] on 10/7/2018, 05:03 AM (206 days ago): <p>We appreciate you taking the time to report this problem. We are currently prioritizing problems that are impacting a broad set of our customers, so we may not be able to investigate this one immediately. We know this problem is important to you, so we will continue to monitor it.</p>

Visual Studio Feedback System on 5/1/2019, 02:22 AM (7 hours ago): <p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

_These are the original issue solutions:_
(no solutions)

**Comment by sharwell** [comment]
Tentatively marking this as **Won't Fix**, as this is not a scenario we would plan to build tests and maintain support for. I will update with a final conclusion following the design review.

**Comment by sharwell** [comment]
This went to a design review 6 May 2019. The team agreed that supporting this scenario would not be maintainable in the long term. However, we're reaching out to the user who reported the issue in hopes of better understanding how it was encountered.

**Comment by sharwell** [comment]
Leaving this open for discussion since the original reporter (@svick) is a GitHub user.

**Comment by svick** [comment]
@sharwell 

> we're reaching out to the user who reported the issue in hopes of better understanding how it was encountered.

What exactly do you want to know?

I was trying to implement my own variation of code generating extensions (https://github.com/dotnet/csharplang/issues/107), which means I needed to override handling of .cs files in VS. This was an early attempt to do that and I was surprised that I got `NullReferenceException`, so I reported it.

In the end, I found another way to do it, so if you made this work, it probably wouldn't help me now. Since it seems you decided that this is an unsupported scenario and presumably that throwing `NullReferenceException` is okay, I am fine with closing this issue.

**Comment by sharwell** [comment]
@svick Thanks, that's what we were looking for! Since we knew the scenario would not be supported, we wanted to see if there was something else we could do to unblock you; it sounds like you are already unblocked.
